src/Elasticsearch5/SearchAdapter/Aggregation/Interval.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace ScandiPWA\CatalogGraphQl\Elasticsearch5\SearchAdapter\Aggregation;
+
+use Magento\Elasticsearch\SearchAdapter\ConnectionManager;
+use Magento\Elasticsearch\Model\Adapter\FieldMapperInterface;
+use Magento\Elasticsearch\Model\Config;
+use Magento\Elasticsearch\SearchAdapter\SearchIndexNameResolver;
+use Magento\CatalogSearch\Model\Indexer\Fulltext;
+use Magento\Elasticsearch\Elasticsearch5\SearchAdapter\Aggregation\Interval as CoreInterval;
+
+/**
+ * Aggregate price intervals for search query result.
+ */
+class Interval extends CoreInterval
+{
+    /**
+     * Minimal possible value
+     */
+    const DELTA = 0.005;
+
+    /**
+     * @var ConnectionManager
+     */
+    private $connectionManager;
+
+    /**
+     * @var FieldMapperInterface
+     */
+    private $fieldMapper;
+
+    /**
+     * @var Config
+     */
+    private $clientConfig;
+
+    /**
+     * @var string
+     */
+    private $fieldName;
+
+    /**
+     * @var string
+     */
+    private $storeId;
+
+    /**
+     * @var array
+     */
+    private $entityIds;
+
+    /**
+     * @var SearchIndexNameResolver
+     */
+    private $searchIndexNameResolver;
+
+    /**
+     * @param ConnectionManager $connectionManager
+     * @param FieldMapperInterface $fieldMapper
+     * @param Config $clientConfig
+     * @param SearchIndexNameResolver $searchIndexNameResolver
+     * @param string $fieldName
+     * @param string $storeId
+     * @param array $entityIds
+     */
+    public function __construct(
+        ConnectionManager $connectionManager,
+        FieldMapperInterface $fieldMapper,
+        Config $clientConfig,
+        SearchIndexNameResolver $searchIndexNameResolver,
+        string $fieldName,
+        string $storeId,
+        array $entityIds
+    ) {
+        $this->connectionManager = $connectionManager;
+        $this->fieldMapper = $fieldMapper;
+        $this->clientConfig = $clientConfig;
+        $this->fieldName = $fieldName;
+        $this->storeId = $storeId;
+        $this->entityIds = $entityIds;
+        $this->searchIndexNameResolver = $searchIndexNameResolver;
+
+        parent::__construct(
+            $connectionManager,
+            $fieldMapper,
+            $clientConfig,
+            $searchIndexNameResolver,
+            $fieldName,
+            $storeId,
+            $entityIds,
+
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function load($limit, $offset = null, $lower = null, $upper = null)
+    {
+
+        $from = ['gte' => 0];
+        $to = ['lt' => 0];
```

Comment:
Please leave a descriptive commend and a link to the issue in order to describe the purpose of this override.
Same on all other methods you overridden, so we can see what actually changed.